pkg/3scale/amp/operator/backend_options_provider.go

```diff
@@ -192,6 +192,12 @@ func (o *OperatorBackendOptionsProvider) listenerPodTemplateLabels() map[string]
 		labels[k] = v
 	}
 
+	if o.apimanager.Spec.Backend.ListenerSpec.Labels != nil {
```

Comment:
this `if` is not needed

Reply:
Addressed in this [commit](https://github.com/3scale/3scale-operator/pull/835/commits/f903bf41523372ad2297594e4efac295a12c0665)